enterprise/tools/upload_local_snapshot/upload_local_snapshot.go

```diff
@@ -0,0 +1,274 @@
+	if err != nil {
+		return nil, status.UnavailableErrorf("failed to read chunked file tree: %s", status.Message(err))
+	}
+	tree := &repb.Tree{}
+	if err := proto.Unmarshal(b, tree); err != nil {
+		return nil, status.WrapError(err, "unmarshall chunked file tree")
+	}
+	return tree, nil
+}
+
+func getMissingDigestsRemoteCache(ctx context.Context, env environment.Env, remoteInstanceName string, digests []*repb.Digest) []*repb.Digest {
+	req := &repb.FindMissingBlobsRequest{
+		InstanceName:   remoteInstanceName,
+		BlobDigests:    digests,
+		DigestFunction: repb.DigestFunction_BLAKE3,
+	}
+	res, err := env.GetContentAddressableStorageClient().FindMissingBlobs(ctx, req)
+	if err != nil {
+		log.Fatalf("Error FindMissingBlobs: %s", err.Error())
+	}
+	return res.MissingBlobDigests
+}
+
+func uploadDigestsRemoteCache(ctx context.Context, env environment.Env, remoteInstanceName string, digests []*repb.Digest) {
+	tmpDir, err := os.MkdirTemp("", "upload-local-ss-*")
+	if err != nil {
+		log.Fatalf("Error making temp dir: %s", err.Error())
+	}
+	defer os.RemoveAll(tmpDir)
+
+	for _, d := range digests {
+		upload := func() {
+			path := filepath.Join(tmpDir, d.GetHash())
+			node := &repb.FileNode{
+				Digest: d,
+			}
+			inFC := env.GetFileCache().FastLinkFile(ctx, node, path)
+			if !inFC {
+				log.Fatalf("Digest %s does not exist in the local file cache", d)
+			}
+
+			rn := digest.NewResourceName(d, remoteInstanceName, rspb.CacheType_CAS, repb.DigestFunction_BLAKE3)
+			rn.SetCompressor(repb.Compressor_ZSTD)
+			file, err := os.Open(path)
+			defer file.Close()
+			if err != nil {
+				log.Fatalf("Error opening file %s: %s", path, err)
+			}
+			_, err = cachetools.UploadFromReader(ctx, env.GetByteStreamClient(), rn, file)
+			if err != nil {
+				log.Fatalf("Error uploading file to remote cache %s: %s", path, err)
+			}
+		}
+
+		upload()
+	}
+}
+
+func uploadManifestRemoteCache(ctx context.Context, env environment.Env, key *fcpb.SnapshotKey, groupID string, remoteInstanceName string) {
+	localManifestACResult := readLocalManifestACResult(ctx, env, key, groupID)
+
+	kd, err := digest.ComputeForMessage(key, repb.DigestFunction_SHA256)
+	if err != nil {
+		log.Fatalf("Error generating digest for snapshot key: %s", err)
+	}
+	remoteManifestKey := &repb.Digest{
+		Hash:      hashStrings(kd.GetHash(), ".manifest"),
+		SizeBytes: 1, /*arbitrary size*/
+	}
+	acDigest := digest.NewResourceName(remoteManifestKey, remoteInstanceName, rspb.CacheType_AC, repb.DigestFunction_BLAKE3)
+	if err := cachetools.UploadActionResult(ctx, env.GetActionCacheClient(), acDigest, localManifestACResult); err != nil {
+		log.Fatalf("Error uploading manifest to remote cache: %s", err)
+	}
+}
+
+func readLocalManifestACResult(ctx context.Context, env environment.Env, key *fcpb.SnapshotKey, groupID string) *repb.ActionResult {
```

Comment:
nit: can we call the func from snaploader instead? just want to avoid the logic going out of sync (these tools tend to break a lot, just want to minimize the amount of maintenance we have to do)

Reply:
Done. Did some refactoring to reuse more logic